### PR TITLE
Update to the new parent POM

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] - Upgrade POM
- [x] - Fix Javadocs
- [x] - Upgrade Tests to the new HtmlUnit version
- [x] - Escape Jelly files by default
- [x] - Address warnings about NPEs

@abayer @reviewbybees

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Findbugs fails, but issues are not being reported in CI

**Comment by oleg-nenashev** [comment]
All findbugs issues seem to be non-critical.
But I want to address them in another PR before the release

File: src/main/java/hudson/plugins/throttleconcurrents/ThrottleQueueTaskDispatcher.java [hunk 1]
```diff
@@ -256,6 +259,11 @@ private boolean isAnotherBuildWithSameParametersRunningOnAnyNode(Queue.Item item
 
     private boolean isAnotherBuildWithSameParametersRunningOnNode(Node node, Queue.Item item) {
         ThrottleJobProperty tjp = getThrottleJobProperty(item.task);
+        if (tjp == null) {
+            // If the property has been ocasionally deleted by this call, 
+            // it does not make sense to limit the throttling by parameter.
+            return false;
+        }
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/throttleconcurrents/ThrottleQueueTaskDispatcher.java]: @balldarrens FYI

File: src/test/java/hudson/plugins/throttleconcurrents/testutils/HtmlUnitHelper.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.List;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * Stores helpers for HtmlUnit.
+ * @author Oleg Nenashev
+ * @since TODO
```
Comment (recena) [inline src/test/java/hudson/plugins/throttleconcurrents/testutils/HtmlUnitHelper.java]: If you expect to cut a release soon, why not specific the version?

**Comment by recena** [comment]
:bee:

File: src/test/java/hudson/plugins/throttleconcurrents/testutils/HtmlUnitHelper.java (hunk 2 above)
Comment (oleg-nenashev) [inline src/test/java/hudson/plugins/throttleconcurrents/testutils/HtmlUnitHelper.java]: I'll fix it during the release

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Comment by oleg-nenashev** [comment]
Going forward as a plugin maintainer

**Pull request merged**